### PR TITLE
Fix/6826

### LINKED ISSUE #6826: PacketFence PKI provider: revoke_on_unregistration setting doesn't work
**Describe the bug**
If you use a "Revoke on unregistration" setting with a PacketFence PKI provider, your certificate is not revoked when device is unregistered.

Error logs:

```
Jan 13 09:32:29 pfel8dev packetfence[3597]: pfperl-api(1858) INFO: getting security_events triggers for accounting cleanup (pf::accounting::acct_maintenance)
Jan 13 09:32:29 pfel8dev packetfence[3598]: pfperl-api(1859) INFO: Instantiate profile default (pf::Connection::ProfileFactory::_from_profile)
Jan 13 09:32:29 pfel8dev packetfence[3598]: pfperl-api(1859) INFO: Request to /api/v1/pki/cert/nqb/1 is unauthorized, will perform a login (pf::api::unifiedapiclient::call)
Jan 13 09:32:29 pfel8dev pfpki[1567]: t=2022-01-13T09:32:29+0000 lvl=eror msg="record not found" pid=1567
```

**To Reproduce**
1. Configure CA and user certificate template in pfpki
2. Configure a PKI provider with certificate provided by MAC
3. Configure a provisioner which use PKI provider
3. Assign provisioner to your connection profile
3. Go on captive portal to generate a certificate and register your device
4. Unregister your device

=> Certificate is not revoked

**Expected behavior**
When node is unregistered, certificate should be revoked.

**Additional context**
I'm not sure `/api/v1/pki/cert/nqb/1` is a valid path for `pfpki`. My certificate was available through `/api/v1/pki/cert/8`

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Fixes revoke on unregister

# Impacts
pki

# Issue
fixes #6826

# Delete branch after merge
YES

# Checklist
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries

## Bug Fixes
* Fixes revoke certificate on unregistration (#6826)

### CODE REVIEW

**Review by nqb — CHANGES_REQUESTED** [review]
fixed: `"my" variable $status masks earlier declaration in same scope at /usr/local/pf/lib/pf/pki_provider/packetfence_pki.pm line 123.`

**Review by nqb — CHANGES_REQUESTED** [review]
- My device is correctly unreg even if my certificate is not found
- My certificate is not found and I got same message:

```
│Jan 14 21:50:36 pfdeb11localdev packetfence[114072]: pfcmd.pl(114072) INFO: Request to /api/v1/pki/cert/26/toto/1 is unauthorized, will perform a login (pf::api::unifiedapiclient::call)                                                  │
│Jan 14 21:50:36 pfdeb11localdev packetfence[114072]: pfcmd.pl(114072) INFO: re-evaluating access (manage_deregister called) (pf::enforcement::reevaluate_access)                                                                           │
│Jan 14 21:50:36 pfdeb11localdev packetfence[114072]: pfcmd.pl(114072) WARN: Can't re-evaluate access because no open locationlog entry was found (pf::enforcement::reevaluate_access)                                                      │
│Jan 14 21:50:36 pfdeb11localdev packetfence[114072]: pfcmd.pl(114072) INFO: modified 00:11:22:33:44:55 from status 'reg' to 'unreg' based on unregdate colum (pf::node::nodes_maintenance)                                                 │
│Jan 14 21:50:36 pfdeb11localdev pfpki[82625]: t=2022-01-14T21:50:36+0000 lvl=eror msg="record not found" pid=82625 
```    

In DB, cert generated by user has id 26.

I pushed some commits to `devel` which prevents code to work properly. I'm rebasing this branch on devel.

File: lib/pf/pki_provider/packetfence_pki.pm [hunk 1]
```diff
@@ -99,28 +95,12 @@ Revoke the certificate for a user
...
 
-    ($status, my $count) = pf::dal::key_value_storage->remove_items(%options);
-    
-    if (is_error($status)) {
-        $logger->error("Unable to delete the key value '$cn'");
+    my $value = eval {
+        my $return = pf::api::unifiedapiclient->default_client->call("DELETE", "/api/v1/pki/cert/$self->{'profile'}/$cn/1");
+    };
+    if ($@) {
+        $logger->debug("Unable to revoke certificate associated with this cn $cn");
```
Comment (nqb) [inline lib/pf/pki_provider/packetfence_pki.pm:103]: Missing `return;` after this line ?

**Review by nqb — CHANGES_REQUESTED** [review]

File: lib/pf/pki_provider/packetfence_pki.pm (hunk 1 above)
Comment (fdurand) [reply lib/pf/pki_provider/packetfence_pki.pm:103]: we don´t test the return value.
The code revoke the cert if it exist and do nothing if it doesn´t exist.

**Pull request merged**